File: app/assets/js/session/containers/new.js
import React from 'react';

import authenticityToken from 'js/_helpers/authenticity_token';

const SessionsNew = React.createClass({
  render() {
    return (
      <div>
        <h1>{'Log in'}</h1>

        <form action='/sessions' acceptCharset='UTF-8' method='post'>
          <input
            type='hidden'
            name='authenticity_token'
            value={authenticityToken()}
          />
          <p>
            <label htmlFor='email'>{'Email'}</label>
            <br />
            <input type='text' name='session[email]' id='email' />
          </p>
          <p>
            <label htmlFor='password'>{'Password'}</label>
            <br />
            <input type='password' name='session[password]' id='password' />
          </p>
          <p className='button'>
            <input type='submit' name='commit' value='Login' />
          </p>
        </form>
      </div>
    );
  },
});

export default SessionsNew;

File: app/views/sessions/new.rb
# frozen_string_literal: true

module Views
  module Sessions
    class New < Views::Base
      def view_template
        form_with(**form_options) do |form|
          h1 { "Log in" }
          field(form, :email) { form.text_field(:email) }
          field(form, :password) { form.password_field(:password) }
          p { form.submit("Login") }
        end
      end

      private

      def form_options
        { scope: :session, url: session_path, class: "login-form" }
      end

      def field(form, name)
        p do
          form.label(name)
          br
          yield
        end
      end
    end
  end
end
